tests/ZendTest/Db/Adapter/Driver/Pgsql/ConnectionTest.php

```diff
@@ -80,4 +80,25 @@ public function testGetConnectionStringEncodeSpecialSymbol()
             $getConnectionString->invoke($this->connection)
         );
     }
+
+    /**
+     * @expectedException \Zend\Db\Adapter\Exception\InvalidArgumentException
+     */
+    public function testSetConnectionTypeException()
+    {
+        $this->connection->setType(3);
+    }
+
+    /**
+     * Test the connection type setter
+     */
+    public function testSetConnectionType()
+    {
+        $type = PGSQL_CONNECT_FORCE_NEW;
+        $this->connection->setType($type);
+        $reflectionProperty = new \ReflectionProperty($this->connection, 'type');
+        $reflectionProperty->setAccessible(true);
+        $this->assertEquals($type, $reflectionProperty->getValue($this->connection));
```

Comment:
Instead of `ReflectionProperty` you can use `readAttribute()` for this:

```php
public function testSetConnectionType()
{
    $type = PGSQL_CONNECT_FORCE_NEW;
    $this->connection->setType($type);
    $this->assertEquals($type, self::readAttribute($this->connection, 'type'));
}
```